src/test/java/com/aws/greengrass/clientdevices/auth/api/DomainEventsTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.api;
+
+import com.aws.greengrass.testcommons.testutilities.GGExtension;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+
+@ExtendWith({MockitoExtension.class, GGExtension.class})
+public class DomainEventsTest {
+    private DomainEvents domainEvents;
+
+    static class TestEvent implements DomainEvent {}
+    static class TestEvent2 implements DomainEvent {}
+
+    @BeforeEach
+    void beforeEach() {
+        domainEvents = new DomainEvents();
+    }
+
+    @Test
+    void GIVEN_noListener_WHEN_eventEmitted_THEN_itsFine() {
+        domainEvents.emit(new TestEvent());
+        domainEvents.emit(new TestEvent2());
+    }
+
+    @Test
+    void GIVEN_twoListenersForSameEvent_WHEN_eventEmitted_THEN_bothListenersFire() {
+        AtomicInteger listener1Count = new AtomicInteger(0);
```

Comment:
Do we need to use an `AtomicInteger` for tests? I get it if we were using multiple threads on the test

Reply:
This isn't to avoid a race. You can only reference effectively final variables in a Lambda function, so this can't be a regular `int`.